src/gas_metering/mod.rs

```diff
@@ -171,47 +181,75 @@ pub fn inject<R: Rules>(
 			elements::Section::Code(code_section) =>
 				for func_body in code_section.bodies_mut() {
 					for instruction in func_body.code_mut().elements_mut().iter_mut() {
-						if let Instruction::Call(call_index) = instruction {
-							if *call_index >= gas_func {
-								*call_index += 1
-							}
+						match instruction {
+							Instruction::GetGlobal(global_index) =>
```

Comment:
Can be one statement.

Could also use guards (https://doc.rust-lang.org/rust-by-example/flow_control/match/guard.html).